### PR TITLE
Fix(UI)#8737: Converted edit description modals into antd modals

### LINKED ISSUE #8737: Mouse clicks on the dark overlay outside of a Modal does not automatically close the Modal
**Affected module**

UI

**Describe the bug**

Clicking with the mouse on the dark overlay outside of a Modal does not automatically close the Modal. This type of behavior is quite common on front-ends and users may expect it, however it does not work. The user must explicitly click on the "Cancel" button. 

**To Reproduce**

Screenshot 2022-11-14 at 4 11 10 PM

**Expected behavior**

Clicking with the mouse on the dark overlay outside of a Modal automatically closes the Modal.

**Version:**
 - OS: OS X
 - Python version:
 - OpenMetadata version: 0.12.2
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by aniketkatkar97**
### Describe your changes :

I worked on converting edit description modals to antd modals
Closes #8737

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">

https://user-images.githubusercontent.com/51777795/208237647-74f0a2d8-c1b8-41c6-a347-88ad40dcb0f5.mov

</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Frontend: @open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/DashboardDetails/DashboardDetails.component.tsx [hunk 1]
```diff
@@ -849,19 +840,6 @@ const DashboardDetails = ({
...
-      {selectedField ? (
-        <RequestDescriptionModal
-          createThread={createThread}
-          defaultValue={getDefaultValue(owner as EntityReference)}
-          header="Request description"
-          threadLink={getEntityFeedLink(
-            EntityType.DASHBOARD,
-            dashboardFQN,
-            selectedField
-          )}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DashboardDetails/DashboardDetails.component.tsx:861]: Why we just removed it? It wasn't needed anymore?

File: openmetadata-ui/src/main/resources/ui/src/components/Modals/ModalWithMarkdownEditor/ModalWithMarkdownEditor.tsx [hunk 2]
```diff
@@ -11,127 +11,81 @@
...
-        <div className="tw-modal-body">
-          <RichTextEditor
-            initialValue={value}
-            placeHolder={placeholder}
-            ref={markdownRef}
-          />
-        </div>
-        <div className="tw-modal-footer">
+    return ReactDOM.createPortal(
+      <Modal
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Modals/ModalWithMarkdownEditor/ModalWithMarkdownEditor.tsx:56]:
```suggestion
    return  <Modal
```

File: openmetadata-ui/src/main/resources/ui/src/components/PipelineDetails/PipelineDetails.component.tsx [hunk 3]
```diff
@@ -1036,19 +1024,6 @@ const PipelineDetails = ({
           onCancel={onThreadPanelClose}
         />
       ) : null}
-      {selectedField ? (
-        <RequestDescriptionModal
-          createThread={createThread}
-          defaultValue={getDefaultValue(owner as EntityReference)}
-          header={t('label.request-description')}
-          threadLink={getEntityFeedLink(
-            EntityType.PIPELINE,
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/PipelineDetails/PipelineDetails.component.tsx:1045]: Why just removed?

File: openmetadata-ui/src/main/resources/ui/src/components/DashboardDetails/DashboardDetails.component.tsx (hunk 1 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/DashboardDetails/DashboardDetails.component.tsx]: It shouldn't have been removed. I'll revert this.

File: openmetadata-ui/src/main/resources/ui/src/components/PipelineDetails/PipelineDetails.component.tsx (hunk 3 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/PipelineDetails/PipelineDetails.component.tsx]: will revert this

**Comment by aniketkatkar97** [comment]
@chirag-madlani, I have pushed the changes as per the comments. Can you please review it?

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**